src/PJ_eqearth.c

```diff
@@ -28,28 +30,59 @@ PROJ_HEAD(eqearth, "Equal Earth") "\n\tPCyl., Sph.";
 #define EPS 1e-11
 #define MAX_ITER 12
 
-static XY s_forward (LP lp, PJ *P) {           /* Spheroidal, forward */
+struct pj_opaque {
+    double  qp;
+    double  rqda;
+    double *apa;
+};
+
+static XY e_forward (LP lp, PJ *P) {           /* Ellipsoidal/Spheroidal, forward */
     XY xy = {0.0,0.0};
-    double phi, phi2, phi6;
-    (void) P;
+    struct pj_opaque *Q = P->opaque;
+    double sbeta;
+    double psi, psi2, psi6;
+
+    if (P->es != 0.0) {
```

Comment:
A little bit of nitpicking here:  For clarity, I would prefer to have block-scoped comments *precede* the block they refer to:

```
/* Ellipsoidal case, converting to authalic latitude */
if (P->es != 0.0) {
    sbeta = pj_qsfn(sin(lp.phi), P->e, 1.0 - P->es) / Q->qp;

    /* Rounding error. */
    if (fabs(sbeta) > 1) 
        sbeta = sbeta > 0 ? 1 : -1;
}

/* Spheroidal case, using latitude */
else 
    sbeta = sin(lp.phi);
```

Or with even less branching:

```
sbeta = sin(lp.phi);

/* In the ellipsoidal case, we convert sbeta to authalic latitude */
if (P->es != 0.0) {
    sbeta = pj_qsfn(sbeta, P->e, 1.0 - P->es) / Q->qp;

    /* Rounding error. */
    if (fabs(sbeta) > 1) 
        sbeta = sbeta > 0 ? 1 : -1;
}
```

Reply:
Sorry Thomas, I merged this while you were writing...

Reply:
Don't worry - although I think my suggestion has some merit in terms of clarity, it also is mostly cosmetical and may be simply a matter of personal preference

Reply:
I can do the change and another PR if you would like to...

Reply:
@beuan: Well - I wrote the comment because *I* would prefer to minimize branching and have block-scope comments precede the block.

But it's your code, and it's already merged, so please only make the change if you personally find my (cosmetical) arguments highly convincing, and would *like* to see the change - in which case I will obviously feel honoured, that you found the comment convincing and/or useful :-)

Reply:
#1102